modules/roles_profiles/manifests/profiles/mac_v3_signing.pp

```diff
@@ -31,16 +31,97 @@
-                    ensure  => 'link',
-                    target  => '/usr/local/bin/python3',
-                    require => Class['packages::python3_s3'],
+            class { 'scriptworker_prereqs': }
+
+            # Template variables for script_config.yaml
+            #   - taskcluster_scope_prefix
+            #   - config file role name
+            #   - sign_with_entitlements
+            #   - verify_mac_signature
+            #   - base_bundle_id
+            #   - identity
+            #   - keychain_password
+            #   - pkg_cert_id
+            #   - notarization_account
+            #   - notarization_password
+            #   - apple_asc_provider
+            #   - notarization_poll_timeout
+            #   - widevine_url
+            #   - widevine_user
+            #   - widevine_pass
+            #   - omnija_url
+            #   - omnija_user
+            #   - omnija_pass
+            #   - langpack_url
+            #   - langpack_user
+            #   - langpack_pass
+            # Template variables for scriptworker.yaml
+            #   - worker_type
+            #   - taskcluster_access_token
+            #   - taskcluster_client_id
+            #   - sign_chain_of_trust
+            #   - verify_chain_of_trust
+            #   - verify_cot_signature
+
+            # TODO Don't create these if the secrets service is unsatisfatory.
+            # Cert files to create
+            #   - dep:
+            #     - widevine_dep.crt from signing_keys.widevine_dep_crt
+            #     - dep_signing.keychain from signing_keys.dep_signing_keychain
+            #  - default
+            #    - widevine_prod.crt
+            #    - nightly_signing.keychain
+            #    - release-signing.keychain
+            #    - ed25519_privkey
+            #
+
+            # For cloning the widevine repository
+            $widevine_user = lookup('widevine_config.user')
+            $widevine_key = lookup('widevine_config.key')
+
+            $role = $facts['networking']['hostname']? {
+                /^dep-mac-v3-signing\d+/ => 'dep',
+                /^tb-mac-v3-signing\d+/ => 'tb-prod',
+                default => 'ff-prod',
             }
-            contain packages::virtualenv_python3_s3
 
+            $worker_config = lookup("signingworker_config.${role}", Hash, undef, {})
+            $role_config = lookup("signingworker_roles.${role}", Hash, undef, {})
+
+            $scriptworker_users = lookup("scriptworker_users.${role}")
+
+            $scriptworker_users.each |String $user, Hash $user_data| {
+                signing_worker { "signing_worker_${user}":
+                    role                => $role,
+                    user                => $user,
+                    password            => lookup("${user}_user.password"),
+                    salt                => lookup("${user}_user.salt"),
+                    iterations          => lookup("${user}_user.iterations"),
```

Comment:
This is perfectly fine.  No need to merge hashes as discussed in slack.  Every time this resource is instantiated over the iteration, you have the keys (role and username) to lookup the secrets under a different but similar data structure in vault.yaml.  The lookup would look like:
```
password => lookup("scriptworker_secrets.${role}.${user}.password").
salt  => lookup("scriptworker_secrets.${role}.${user}.salt").
iterations => lookup("scriptworker_secrets.${role}.${user}.iterations").
```

and the corresponding secrets nested hash would look like this:
```
scriptworker_secrets:
  dep:
    depbld1:
		password:
		salt:
		iteration:
    depbld1:
		password:
		salt:
		iteration:
    tbbld:
		password:
		salt:
		iteration:
  tb-prod:
    cltbld:
		password:
		salt:
		iteration:
  ff-prod:
    cltbld:
		password:
		salt:
		iteration:
```

Reply:
Another reason not to attempt merging hashes while in hiera is, Vault (real vault, not vault.yaml) will be under a unique prefix namespace to signify the lookup is reaching into vault and not the in-tree yaml files.  A vault lookup will look like this:

` password => lookup("vault_secrets.scriptworker_secrets.${role}.${user}.password").`

Reply:
This is not the section from yesterday's discussion about merging, it's the worker and role config from lines 99 and 100. These are used to fill out templates, which have a variable number of sections in depending on the role the node is performing. See https://github.com/mozilla-platform-ops/ronin_puppet/pull/73/files/c5e4b0374a5531c3238613c2c89b48bf731401e7#diff-01a7e7a65760ec28c6ead2e535a11db6R17 for its current use.